### ISSUE TITLE
Incorrect format generated for date

### ISSUE THREAD

**Issue by we-zhang** [bshamblen/json-schema-generator#4]
The generated format is "date-time" for instance "2015-01-01" which is incorrect, it should be "date".

JSON Sample:
```
{
  "dt": "2015-01-01", 
  "dt_time": "2012-04-23T18:25:43.511Z"
}
```

JSON Schema:
```
{
   "type": "object",
   "properties": {
      "dt": {
         "id": "dt",
         "type": "string",
         "format": "date-time",
         "default": "2015-01-01"
      },
      "dt_time": {
         "id": "dt_time",
         "type": "string",
         "format": "date-time",
         "default": "2012-04-23T18:25:43.511Z"
      }
   }
}
```

**Comment by bshamblen** [comment]
The JSON Schema spec only has a `date-time` format defined. 

http://json-schema.org/latest/json-schema-validation.html#rfc.section.7.3

**Comment by we-zhang** [comment]
@bshamblen Yes, the spec doesn't list 'date' format, but as per documentation http://json-schema.org/latest/json-schema-validation.html#rfc.section.7.2, custom format attribute is allowed, but this need the agreement in the validator/implementation. 

Now the problem is since there's `date-time` format generated for instance `2015-01-01`, most of validator will check this format and validate against the value, it will be failed since it's not a date-time.

I think if we don't add custom format attribute, then it's better to not generate the date-time format for date value.

**Comment by bshamblen** [comment]
@weching I understand the desire to break it down further, as many things that leverage JSON schema do. However, I could see the argument for supporting both. 

The original reason we created this package was to be able to take JSON data and generate schemas that use a proprietary schema definition syntax, which was loosely based on JSON schema. We then used those schemas to generate SimpleSchemas, to dynamically generate forms using AutoForm, and API documentation using Swagger.

Since `date` is not explicitly included in the JSON schema spec, returning `date` *could* break *some* tools that don't support that `format`. Not returning `date` *could* break others that have chosen to implement that `format`.

The [Open API spec](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#data-types) (formerly Swagger) supports a `string` data type, with a `format` of either `date` and `date-time`. 

[SimpleSchema](https://github.com/aldeed/meteor-simple-schema#type), on the other hand, only supports a `Date` type.

I would lean toward making this another option, so users could choose the desired behavior based on their need.
